x-pack/plugins/observability/server/services/slo/sli_client.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+import { SLO_DESTINATION_INDEX_NAME } from '../../assets/constants';
+import { InternalQueryError, NotSupportedError } from '../../errors';
+import { IndicatorData, SLO } from '../../types/models';
+
+export interface SLIClient {
+  fetchDataForSLOTimeWindow(slo: SLO): Promise<IndicatorData>;
+}
+
+export class DefaultSLIClient implements SLIClient {
+  constructor(private esClient: ElasticsearchClient) {}
+
+  async fetchDataForSLOTimeWindow(slo: SLO): Promise<IndicatorData> {
+    if (slo.budgeting_method !== 'occurrences') {
+      throw new NotSupportedError(`Budgeting method: ${slo.budgeting_method}`);
+    }
+
+    const result = await this.esClient.search({
+      size: 0,
+      index: `${SLO_DESTINATION_INDEX_NAME}*`,
+      query: {
+        bool: {
+          filter: [{ term: { 'slo.id': slo.id } }, { term: { 'slo.revision': slo.revision } }],
+        },
+      },
+      aggs: {
+        full_window: {
+          date_range: {
+            field: '@timestamp',
+            ranges: [fromSLOTimeWindowToRange(slo)],
+          },
+          aggs: {
+            good: { sum: { field: 'slo.numerator' } },
```

Comment:
nit: just of curiosity, is `good` the term that should go with it? I'm not against it. e.g. why not use `numerator`

Reply:
SLO literatures define an SLI as the ratio of the good over total events. They are used interchangeably. I prefer `good` and `total` as opposed to `numerator` and `denominator`, because the former speaks the business whereas the later speaks implementation details of the calculus.